feat: Initial Death on the Isle quest helper

pajlada opened the pull request:
- [x] Is the difficulty correct?
 - [x] Are the skill requirements correct?
 - [ ] Is the position in the optimal quest guide & optimal ironman quest guide correct?
       No
 - [x] Are the monsters you have to fight correctly listed?
 Yes, although the wording is open for change
 - [x] Are the rewards correct?
 - [x] Does the "RuneScape Wiki" link work?

Zoinkwiz commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+		interrogatePavoAgain = new NpcStep(this, NpcID.PAVO, new WorldPoint(1442, 2931, 2), "Interrogate Pavo.");
+		interrogatePavoAgain.addDialogStep("No.");
+
+		accuseAdala = new NpcStep(this, NpcID.ADALA, new WorldPoint(1446, 2933, 2), "Accuse Adala of the crime, ready for an easy fight. You can use Protect from Melee to negate all damage.");
```
Fun thing I found, if you hit 1hp you say you yield and the quest continues, so seems safe.

pajlada replied:
Removed any mention of protect from melee in [`900ec8b` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/900ec8b749f35206cf1ccc3b4b53fd58cf7efc39)

Zoinkwiz commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+		// NOTE: The stairs don't highlight well. A tile step would be better, but tile steps don't work here.
+		var headDownFromTopFloor = new ObjectStep(this, ObjectID.STAIRCASE_54714, new WorldPoint(1445, 2939, 2), "Climb down the staircase.");
+		var headDownFromMiddleFloor = new ObjectStep(this, ObjectID.STAIRCASE_54714, new WorldPoint(1442, 2936, 1), "Climb down the staircase.");
+		var climbFirstLooseRocksToTheatre = new ObjectStep(this, ObjectID.LOOSE_ROCKS_54720, new WorldPoint(1469, 2918, 0), "Climb the Loose rocks on your way to the theatre.");
```
Would be good to specify where the rocks are, like 'Climb the loose rocks south-east of the villa on your way to the theatre.'

pajlada replied:
:+1: Fixed in [`8cff659` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/8cff659f0980474a41ee6e5db06c656dc5f7ab5e)

Zoinkwiz commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+
+		// 42 + 45
+		talkToStradiusToEnterTheTheatre = new NpcStep(this, NpcID.STRADIUS, new WorldPoint(1472, 2925, 0), "Talk to Stradius to enter the theatre.");
+		confrontNaiatli = new NpcStep(this, NpcID.NAIATLI, new WorldPoint(1465, 2932, 0), "Confront Naiatli, attacking her when she runs away. When Clodius appears, attack him instead. Use Protect from Melee to avoid all damage.");
```
From what I could tell they never get to attack you, so no need for protect from melee? Not sure if maybe I had 1 strength I wouldn't hit enough to instant-transition maybe, but the attacks felt scripted.

pajlada replied:
The attacks did feel very scripted to me too - only the Adala "fight" took more than one hit

pajlada replied:
I removed any mention of Protect from Melee in the Naiatla/Clodius fight in [`b79e84b` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/b79e84bfc69258b4460cbea6d19b4e919610d035)

pajlada replied:
Changed the combat requirements to just say "2 unlosable fights"
You do get combat experience, so I don't want to remove the combat requirements entirely, and I'm not sure I'm confident enough to say: "if you turn off auto retaliate you can avoid getting combat experience"

Zoinkwiz commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+		// 42 + 45
+		talkToStradiusToEnterTheTheatre = new NpcStep(this, NpcID.STRADIUS, new WorldPoint(1472, 2925, 0), "Talk to Stradius to enter the theatre.");
+		confrontNaiatli = new NpcStep(this, NpcID.NAIATLI, new WorldPoint(1465, 2932, 0), "Confront Naiatli, attacking her when she runs away. When Clodius appears, attack him instead. Use Protect from Melee to avoid all damage.");
+		var killClodius = new NpcStep(this, NpcID.CLODIUS, new WorldPoint(1469, 2937, 0), "Kill Clodius. Use Protect from Melee to avoid all damage.");
```
You just ever attack him, so I think this should be 'Attack Clodius.'

pajlada replied:
Changed phrasing to "Attack" instead of "Kill" in [`5e7e68a` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/5e7e68a2bccbce7f52739851eb498ada919e4e5e)

Zoinkwiz commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+	public List<String> getCombatRequirements()
+	{
+		return List.of(
+			"2 level 49 criminals without gear"
```
Neither seemed to have levels, and you don't actually need to attack one of them. The other doesn't give experience and it doesn't count as your actual hit on them. I'm not sure the best way to convey this, but they're basically not actually enemies.

pajlada replied:
Maybe "2 theatrical fights"?

pajlada replied:
Seems like you can't lose any of the fights actually. On my last playthrough, I'll try just planking to Adala too, and if that's completely safe & continues the quest like normal, I'll remove the combat requirements & any mention of protect from melee

Zoinkwiz reviewed (commented):
Couple points but generally really good, and nice clean code which made it a nice review!

Haavardaw commented:
1. Maybe we should recommend Stamina pots as a standard for quests as it is pretty nice having it added in the bank filter aswell, maybe just say 1x or more Stamina Potions

2. Could do recommend combat gear instead of a specific level, I would say that the enemies to defeat below with levels should already indicate that you want some stats but not required to be maxed. This way we would also have it in the bank filters.

3. On the step "Talk to Patzi at the entrance..." in start of quest and anywhere else you need to get to the city. We should add a recommended teleport of con, con trim, max cape, aldarin teleport tab. On the recommended section and each step and MAYBE a tooltip or just highlight on it.
3.1 Question - Are we able to check where a player has house with some type of varbit or no? if so, we would want to add that there.

4. On first step with "Talk to Constantinius, Cozyac, Pavo, Xocotla" helper tells you to speak to them in reverse order which does look a little weird on the sidebar. I know order you speak to them does not matter but would be just a little nit on that. 
4.1 Similar thing to last one on pickpocketing part tells you to "Pickpocket Adala, Pavo, Cozyac, Xocotla" in that order. Just correct sidebar on both of these.
 
5. "Talk to the Customer" should be "Talk to the Costumer"

6. It rewards 2 quest points, not 5 as shown in the helper.

Great helper and good job with this! All the feedback is just small nits that hopefully help a little with the release.

pajlada commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+		talkToPatziAfterStealingUniform.addSubSteps(continueTalkingToPatzi);
+
+		/// 10 + 12
+		equipButlersOutfitAndHeadInside = new NpcStep(this, NpcID.HEAD_BUTLER, new WorldPoint(1426, 2919, 0), "Equip the Butler's uniform pieces and talk to the Head Butler up the stone stairs and to the south.", uniformEquipped);
```
:+1: I've thrown a tooltip on the "equipped" variant of it, so whenever the user is prompted to talk to the butler to enter, they'll see a tooltip if they don't have it.

Fixed in [`14ed9d8` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/14ed9d88a1b49dd314ee1a26b60620b7b2e312c9)

pajlada commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+		investigateMurder2.addStep(and(pickpocketedItems, not(inspectedWineLabels)), inspectWineLabels);
+		investigateMurder2.addStep(and(pickpocketedItems, not(inspectedThreateningNote)), inspectThreateningNote);
+		investigateMurder2.addStep(and(pickpocketedItems, not(inspectedDrinkingFlask)), inspectDrinkingFlask);
+		investigateMurder2.addStep(and(pickpocketedItems, not(inspectedShippingContract)), inspectShippingContract);
```
Fixed in [`14a4cac` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/14a4cac58e8f6a2e0281abdb473747fce3f75b5f)

We start by checking if a user is missing an item, and if they do, we tell them to pickpocket the relevant guest.
If the user has all items, it will fall through to checking if they've inspected the items. Which will then fall back to having them return to the guards.
I believe this should work as expected

pajlada commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+
+		// varbit 11233: 0 -> 1 when handing over clues to the guards
+
+		investigateMurder2 = new ConditionalStep(this, returnToTheGuards2);
```
Resolved in the comment below

pajlada commented on src/main/java/com/questhelper/helpers/quests/deathontheisle/DeathOnTheIsle.java:
```diff
@@ -0,0 +1,734 @@
+		interrogateCozyacAgain = new NpcStep(this, NpcID.COZYAC, new WorldPoint(1442, 2929, 2), "Interrogate Cozyac.");
+		interrogateCozyacAgain.addDialogStep("No.");
+		interrogatePavoAgain = new NpcStep(this, NpcID.PAVO, new WorldPoint(1442, 2931, 2), "Interrogate Pavo.");
+		interrogatePavoAgain.addDialogStep("No.");
```
Yeah you don't have to press "No." - I never did.
I think I'd prefer to just remove the "No." suggestion altogether, what do you think?

pajlada commented:
> 1. Maybe we should recommend Stamina pots as a standard for quests as it is pretty nice having it added in the bank filter aswell, maybe just say 1x or more Stamina Potions

For this quest, you can't bring anything into the Villa. So in this particular case, I don't think a Stamina potion is necessary. You'd only be able to use it in the first 3-4 steps where you steal the butler's outfit

> 2. Could do recommend combat gear instead of a specific level, I would say that the enemies to defeat below with levels should already indicate that you want some stats but not required to be maxed. This way we would also have it in the bank filters.

No combat gear can be brought in. I took the combat level requirement from the wiki guide, but it seems like you can't lose any of the fights so I've removed the combat level requirement in [`88c4840` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/88c48401b473b96037b5eaf6d7b3d39a0dbaad69)

> 3. On the step "Talk to Patzi at the entrance..." in start of quest and anywhere else you need to get to the city. We should add a recommended teleport of con, con trim, max cape, aldarin teleport tab. On the recommended section and each step and MAYBE a tooltip or just highlight on it.

I'm not sure I like teleport recommendations as the first step. I included it in the text anyway, but I think that's only really truly helpful in the beginning of the release - once a few months have passed, I think people will be able to find their way to Aldarin just with the mention of "Villa Lucens on the island of Aldarin". Not a hill I'll die on, that's just my vibe

> 3.1 Question - Are we able to check where a player has house with some type of varbit or no? if so, we would want to add that there.

Not sure, but I think that would be super handy if we want to create some proper teleport routes for people in the future. The "Shortest route" plugin probably has this feature if it exists.

> 4. On first step with "Talk to Constantinius, Cozyac, Pavo, Xocotla" helper tells you to speak to them in reverse order which does look a little weird on the sidebar. I know order you speak to them does not matter but would be just a little nit on that.

Good catch - fixed in [`ebdedc7` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/ebdedc7863db4c2793d6630e4f49c5a26e83f07e)

> 4.1 Similar thing to last one on pickpocketing part tells you to "Pickpocket Adala, Pavo, Cozyac, Xocotla" in that order. Just correct sidebar on both of these.

I think the order is correct now since a previous fix I've made - I've at least confirmed it starts with Adala

> 5. "Talk to the Customer" should be "Talk to the Costumer"

Fixed in [`38dac24` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/38dac24541444f72fc558c893e8494740add923c) - thanks!

> 6. It rewards 2 quest points, not 5 as shown in the helper.

Fixed in [`df1328b` (#1774)](https://github.com/Zoinkwiz/quest-helper/pull/1774/commits/df1328bfd4f9ad1e9fbbf4e08df7d08763f33f49) - I copied the While Guthix Sleeps helper as the template & just left its quest point reward :sunglasses: 

> Great helper and good job with this! All the feedback is just small nits that hopefully help a little with the release.

:heart:
